junit-platform-engine/src/main/java/org/junit/platform/engine/discovery/DiscoverySelectors.java

```diff
@@ -937,4 +942,44 @@ public static IterationSelector selectIteration(DiscoverySelector parentSelector
 		return new IterationSelector(parentSelector, iterationIndices);
 	}
 
+	/**
+	 * Parse the supplied string representation of a
+	 * {@link DiscoverySelectorIdentifier}.
+	 *
+	 * @param identifier the string representation of a {@code DiscoverySelectorIdentifier};
+	 * never {@code null} or blank
+	 * @since 1.11
+	 * @see DiscoverySelectorIdentifierParser
+	 */
+	@API(status = EXPERIMENTAL, since = "1.11")
+	public static Optional<? extends DiscoverySelector> parse(String identifier) {
+		return new DiscoverySelectorIdentifierParsers().parse(identifier);
```

Comment:
Creating a new instance of `DiscoverySelectorIdentifierParsers` here which leads to a new `ServiceLoader` lookup and instantiation does not feel right here. I don't think we should make it `static`, though. Thoughts?

Reply:
Is this really a problem? I agree that making it `static` feels dirty.
 
Maybe we should add a note to prefer using either `parseAll` when processing more than one, and/or consider making `DiscoverySelectorIdentifierParsers` public so that its `parse` method can be used repeatedly, e.g., when working with streams.

Reply:
I've changed it to be lazily load them only once via a singleton in 9967df695015b9b81a85127f7ce2c854965bccd2